core/dbt/parser/schemas.py

```diff
@@ -878,12 +898,68 @@ def parse_patch(self, block: TargetBlock[NodeTarget], refs: ParserRef) -> None:
                 f"file {source_file.path.original_file_path}"
             )
+
+        # patch versioned nodes
+        versions = block.target.versions
+        if versions:
+            latest_version = block.target.latest_version or max(
+                [version.name for version in versions]
+            )
+            for unparsed_version in versions:
+                versioned_model_name = (
+                    unparsed_version.defined_in or f"{patch.name}_v{unparsed_version.name}"
+                )
+                # ref lookup without version - version is not set yet
+                versioned_model_unique_id = self.manifest.ref_lookup.get_unique_id(
+                    versioned_model_name, None, None
+                )
+                if versioned_model_unique_id is None:
+                    warn_or_error(
+                        NoNodeForYamlKey(
+                            patch_name=versioned_model_name,
+                            yaml_key=patch.yaml_key,
+                            file_path=source_file.path.original_file_path,
+                        )
+                    )
+                    continue
+                versioned_model_node = self.manifest.nodes.pop(versioned_model_unique_id)
+                versioned_model_node.unique_id = (
+                    f"model.{patch.package_name}.{patch.name}.v{unparsed_version.name}"
+                )
+                versioned_model_node.fqn[-1] = patch.name  # bleugh
+                versioned_model_node.fqn.append(f"v{unparsed_version.name}")
+                self.manifest.nodes[versioned_model_node.unique_id] = versioned_model_node
+
+                # flatten columns based on include/exclude
+                version_refs: ParserRef = ParserRef.from_version(
+                    block.target.columns, unparsed_version
+                )
+
+                versioned_model_patch = ParsedNodePatch(
+                    name=patch.name,
+                    original_file_path=versioned_model_node.original_file_path,
+                    yaml_key=patch.yaml_key,
+                    package_name=versioned_model_node.package_name,
+                    description=unparsed_version.description or versioned_model_node.description,
+                    columns=version_refs.column_info,
+                    meta=unparsed_version.meta or versioned_model_node.meta,
+                    docs=unparsed_version.docs or versioned_model_node.docs,
+                    config=unparsed_version.config,
+                    access=versioned_model_node.access,
+                    version=unparsed_version.name,
+                    is_latest_version=latest_version == unparsed_version.name,
+                )
+                versioned_model_node.patch(versioned_model_patch)
+                self.patch_node_config(versioned_model_node, versioned_model_patch)
+            self.manifest.rebuild_ref_lookup()  # TODO: is this necessary?
```

Comment:
I wouldn't think you would need to rebuild the whole thing. There is a self.manifest.ref_lookup.add_node(...) method if you need to add the versioned nodes. It's not clear to me whether that would be necessary or not...